### PR TITLE
Sdit 883 awards

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt [hunk 1]
```diff
@@ -627,11 +644,65 @@ class AdjudicationService(
...
+
+    val offenderBookId = party.offenderBooking!!.bookingId // the adjudication party will always have a booking Id
+
+    val sanctionSeq =
+      adjudicationHearingResultAwardRepository.getNextSanctionSequence(offenderBookId = offenderBookId)
+
+    // latest result should always be target result
+    val hearingResult =
+      adjudicationHearingResultRepository.findFirstOrNullByIncidentChargeOrderById_resultSequenceDesc(
+        party.charges.single(),
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt:671]: I think here, don't we need to latest result for a charge? In other words I suspect we also need to charge sequence for migrated adjudications where there are multiple charges

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultAwardResourceIntTest.kt [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          org.mockito.kotlin.check {
+            assertThat(it).containsEntry("bookingId", prisoner.latestBooking().bookingId.toString())
+            assertThat(it).containsEntry("hearingId", existingHearing.id.toString())
+            assertThat(it).containsEntry("adjudicationNumber", existingAdjudicationNumber.toString())
+            assertThat(it).containsEntry("resultSequence", "1")
+            assertThat(it).containsEntry("sanctionSequence", existingCharge.id.chargeSequence.toString())
+          },
+          isNull(),
+        )
+      }
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultAwardResourceIntTest.kt:210]: I think we could also do a repository read and check that data on the database is correct

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt [hunk 3]
```diff
@@ -222,6 +224,7 @@ class AdjudicationsHearingResultsResourceIntTest : IntegrationTestBase() {
             assertThat(it).containsEntry("hearingId", existingHearing.id.toString())
             assertThat(it).containsEntry("adjudicationNumber", existingAdjudicationNumber.toString())
             assertThat(it).containsEntry("resultSequence", "1")
+            assertThat(it).containsEntry("chargeSequence", existingCharge.id.chargeSequence.toString())
           },
           isNull(),
         )
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationsHearingResultsResourceIntTest.kt:230]: same here, a read of the database would be good to assert on

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt (hunk 1 above)
Comment (lbennett-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt:671]: yes - will add

**Pull request merged**